app/controllers/application_controller.rb

```diff
@@ -348,8 +348,10 @@ def generic_exception(exception)
     if exception.try(:cause).is_a?(ActiveRecord::SubclassNotFound)
       sti_clean_up(exception.cause)
     else
-      Foreman::Logging.exception("Action failed", exception)
-      render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception => exception}
+      ex_message = exception.message
+      Foreman::Logging.exception(ex_message, exception)
+      full_request_id = request.request_id
+      render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception_message => ex_message, request_id: full_request_id.split('-').first, full_request_id: full_request_id}
```

Comment:
`full_request_id.split('-').first` logic is duplicated here with `MultilineRequestPatternLayout`. I wonder if there's a way to avoid that and have a `request.short_request_id`.